platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -148,6 +143,58 @@ def initialize_components_list(self):
         self._component_name_list.append(COMPONENT_CPLD1)
         self._component_name_list.append(COMPONENT_CPLD2)
 
+    ##############################################
+    # SFP methods
+    ##############################################
+    def get_num_sfps(self):
+        """
+        Retrieves the number of sfps available on this chassis
+
+        Returns:
+            An integer, the number of sfps available on this chassis
+        """
+        if self.sfp_module is None:
```

Comment:
suggest adding a dedicated flag for sfp init, instead of using this self.sfp_module.

Reply:
fixed.  an explicit flag which represents whether SFP module initialized has been introduced.